### PR TITLE
Ensure consistent link underline and color on hover

### PR DESCRIPTION
**Pull request by m158261**
## References

This references Text Issue Area # 2 in https://github.com/jupyterlab/jupyterlab/issues/9399

This is a more up to date implementation of https://github.com/jupyterlab/jupyterlab/pull/14914

## Code changes

Changes to CSS to make links more consistent throughout the application. Added new global CSS variables to light and dark themes.

## User-facing changes

Links will have consistent colour and underline on hover therefore improving accessibility.

Current dark theme link

Main branch link dark theme

Updated dark theme link

Link dark theme

Current hover over in dark theme

Main branch link hover over dark theme

Updated hover over in dark theme

Link hover over dark theme

Current light theme link

Main branch link light theme

Updated light theme link

Link light theme

Current hover over in light theme

Main branch link hover over light theme

Updated hover over in light theme

Link hover over light theme

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by m158261** [comment]
Hey @krassowski, I'm getting test failures in galata but different tests are failing each time. Is there anything I can do to make the UI tests more stable for this PR?

**Review by krassowski — APPROVED** [review]
Thank you @m158261. I restated the tests again and hope these will pass now.

if needed we can adjust specific colours later easily thanks to the new variables.

File: packages/extensionmanager/style/base.css
Comment (fcollonval) [inline packages/extensionmanager/style/base.css:51]:
```suggestion
  color: var(--jp-content-link-hover-color, var(--jp-content-link-color));
```

File: packages/extensionmanager/style/base.css
Comment (fcollonval) [inline packages/extensionmanager/style/base.css:192]:
```suggestion
  color: var(--jp-content-link-visited-color, var(--jp-content-link-color));
```

File: packages/extensionmanager/style/base.css
Comment (fcollonval) [inline packages/extensionmanager/style/base.css:197]:
```suggestion
  color: var(--jp-content-link-hover-color, var(--jp-content-link-color));
```

File: packages/rendermime/style/base.css
Comment (fcollonval) [inline packages/rendermime/style/base.css:41]:
```suggestion
  color: var(--jp-content-link-hover-color, var(--jp-content-link-color));
```

File: packages/rendermime/style/base.css
Comment (fcollonval) [inline packages/rendermime/style/base.css:46]:
```suggestion
  color: var(--jp-content-link-visited-color, var(--jp-content-link-color));
```

File: packages/rendermime/style/base.css
Comment (fcollonval) [inline packages/rendermime/style/base.css:265]:
```suggestion
  color: var(--jp-content-link-hover-color, var(--jp-content-link-color));
```

File: packages/rendermime/style/base.css
Comment (fcollonval) [inline packages/rendermime/style/base.css:270]:
```suggestion
  color: var(--jp-content-link-visited-color, var(--jp-content-link-color));
```

**Review by fcollonval — APPROVED** [review]
Thanks @m158261 

I'll push a commit that adds the previous variable as fallback to support existing themes not defining the new variables.

> Similar to what was done in https://github.com/jupyterlab/jupyterlab/pull/11264

**Comment by krassowski** [comment]
@m158261 just FYI, I would have merged it now but I cannot because the CI is running after you merged the main branch in:

image

I appreciate the effort to ensure the changes are still neatly working with the latest HEAD but when folks do that on a number of PRs we simply run out of CI time (there is a number on concurrent jobs we get as open source project) and it sometimes slows down the process.

Not a big problem, just something for you to consider how often you do the merges, especially when the PR is waiting for review when we see a recent activity it may be wrongly interpreted as "still in progress" and makes the maintainer think "I will come back once CI finishes to check the status".

**Comment by tonyfast** [comment]
psyched this is merged. this is a win. thanks y'all!

**Pull request merged**
